test/reporting-test.js

```diff
@@ -134,5 +144,36 @@ describe('reporting.js', () => {
 
             assert.strictEqual(report.clean, false);
         });
+
+        it('should not cache if a scan failed with an exit code that should be ignored', async () => {
```

Comment:
any chance we can test that it DOES cache in the opposite case?

Reply:
Sure

Reply:
fab, it always feels like we're only telling half the story if we only test when it doesn't happen :).